C++ warning on unary operations

It looks like `UnaryOp` in `unary_op_utils.h` needs a specialization for boolean types, to use logical negation.

```
./cunumeric/unary/unary_op_util.h:651:53: warning: bitwise negation of a boolean expression always evaluates to 'true'; did you mean logical negation? [-Wbool-operation]
  constexpr T operator()(const T& x) const { return ~x; }
                                                    ^~
                                                    !
cunumeric/unary/unary_op.cc:42:63: note: in instantiation of member function 'cunumeric::UnaryOp<cunumeric::UnaryOpCode::INVERT, BOOL_LT>::operator()' requested here
      for (size_t idx = 0; idx < volume; ++idx) outptr[idx] = func(inptr[idx]);
                                                              ^
./cunumeric/unary/unary_op_template.inl:67:5: note: in instantiation of member function 'cunumeric::UnaryOpImplBody<cunumeric::VariantKind::CPU, cunumeric::UnaryOpCode::INVERT, BOOL_LT,
      1>::operator()' requested here
    UnaryOpImplBody<KIND, OP_CODE, CODE, DIM>()(func, out, in, pitches, rect, dense);
    ^
/Users/rohany/Documents/nvidia/legate.core/install/include/core/utilities/dispatch.h:30:27: note: in instantiation of function template specialization
      'cunumeric::UnaryOpImpl<cunumeric::VariantKind::CPU, cunumeric::UnaryOpCode::INVERT>::operator()<BOOL_LT, 1, nullptr>' requested here
        return f.template operator()<LegateTypeCode::BOOL_LT, DIM>(std::forward<Fnargs>(args)...);
                          ^
./cunumeric/unary/unary_op_util.h:127:25: note: in instantiation of function template specialization
      'cunumeric::UnaryOpDispatch<cunumeric::VariantKind::CPU>::operator()<cunumeric::UnaryOpCode::INVERT>' requested here
      return f.template operator()<UnaryOpCode::INVERT>(std::forward<Fnargs>(args)...);
                        ^
cunumeric/unary/unary_op.cc:106:3: note: in instantiation of function template specialization 'cunumeric::unary_op_template<cunumeric::VariantKind::CPU>' requested here
  unary_op_template<VariantKind::CPU>(context);
```

--- a/src/cunumeric/unary/unary_op_util.h
+++ b/src/cunumeric/unary/unary_op_util.h
@@ -643,8 +643,9 @@ struct UnaryOp<UnaryOpCode::IMAG, CODE> {
 
 template <legate::LegateTypeCode CODE>
 struct UnaryOp<UnaryOpCode::INVERT, CODE> {
-  static constexpr bool valid = legate::is_integral<CODE>::value;
-  using T                     = legate::legate_type_of<CODE>;
+  static constexpr bool valid =
+    legate::is_integral<CODE>::value && CODE != legate::LegateTypeCode::BOOL_LT;
+  using T = legate::legate_type_of<CODE>;
 
   UnaryOp(const std::vector<legate::Store>& args) {}
 

